src/main/java/redis/clients/util/Pool.java

```diff
@@ -48,6 +49,11 @@ public T getResource() {
     try {
       return internalPool.borrowObject();
     } catch (NoSuchElementException nse) {
+      if (null == nse.getCause()) { // The exception was caused by an exhausted pool
```

Comment:
The client can use the same code snippet by himself but first, the client should be aware of it by digging up to the inner use of `commons-pool2` and the way it distinguish between the two scenarios. 
I bump into it by accident, while examining a Jedis's monitoring failure in my own code and going through `commons-pool2` implementation.
I think that since Jedis's code catches and rethrows the customized exception, it should be specific on the cause and shouldn't be assume that the client aware of its inner use of third parties.
I'll add another unit test for the other scenario, if this still relevant.